Signal/src/ViewControllers/MediaPageViewController.swift

```diff
@@ -4,59 +4,55 @@
 
 import UIKit
 
-public struct MediaGalleryPage: Equatable {
-
-    public let viewController: MediaDetailViewController
-    public let galleryItem: MediaGalleryItem
-
-    public var message: TSMessage {
-        return galleryItem.message
+// Objc wrapper for the MediaGalleryItem struct
+@objc
+public class GalleryItemBox: NSObject {
+    public let value: MediaGalleryItem
+    
+    init(_ value: MediaGalleryItem) {
+        self.value = value
     }
-
+    
     public var attachmentStream: TSAttachmentStream {
-        return galleryItem.attachmentStream
-    }
-
-    public var isVideo: Bool {
-        return galleryItem.isVideo
+        return value.attachmentStream
     }
+}
 
-    public var image: UIImage {
-        // TODO cache this
-        return galleryItem.fullSizedImage
+fileprivate class Box<A> {
+    var value: A
+    init(_ val: A) {
+        self.value = val
     }
+}
 
-    // MARK: Equatable
-
-    public static func == (lhs: MediaGalleryPage, rhs: MediaGalleryPage) -> Bool {
-        return lhs.galleryItem == rhs.galleryItem
+fileprivate extension MediaDetailViewController {
+    fileprivate var galleryItem: MediaGalleryItem {
+        return self.galleryItemBox.value
     }
 }
 
 class MediaPageViewController: UIPageViewController, UIPageViewControllerDataSource, UIPageViewControllerDelegate, MediaDetailViewControllerDelegate {
 
     private weak var mediaGalleryDataSource: MediaGalleryDataSource?
 
-    private var cachedPages: [MediaGalleryPage] = []
```

Comment:
Since a MediaGalleryPage has a strong reference to a MediaDetailViewController, this cache was getting large fast. Since we were treating it as a sort of DataSource as well as a cache, it wasn't convenient to implement caching with the existing layout.

Moving everything around left the MediaGalleryPage struct not very useful, so I nixed it.

Reply:
Can you please ensure that all of the new VCs related to media views log their deallocation (like `OWSViewController`) and verify that none are leaking?

Reply:
Yup, they all include:

```
deinit {
        Logger.debug("\(logTag) deinit")
   }
```